packages/blade/src/components/BottomSheet/__tests__/BottomSheet.web.test.tsx

```diff
@@ -0,0 +1,310 @@
+/* eslint-disable no-global-assign */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React from 'react';
+import userEvents from '@testing-library/user-event';
+import { mockViewport } from 'jsdom-testing-mocks';
+import { within } from '@testing-library/react';
+import { BottomSheet, BottomSheetHeader, BottomSheetBody, BottomSheetFooter } from '../BottomSheet';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+import { Text } from '~components/Typography';
+import { Button } from '~components/Button';
+import { Dropdown } from '~components/Dropdown';
+import { SelectInput } from '~components/Input/SelectInput';
+import { ActionList, ActionListItem } from '~components/ActionList';
+import { Badge } from '~components/Badge';
+import { Counter } from '~components/Counter';
+
+export const sleep = (delay = 10): Promise<number> =>
```

Comment:
Can we rely on testing library primitives for [waiting](https://testing-library.com/docs/guide-disappearance/#waiting-for-appearance)?

Reply:
Tried those none of it worked actually, not exactly sure why. I have to wait for the animations to finish in realtime timers, cant also use jest.useFakeTimers()

Reply:
Why can't we use fakeTimers? For animations its best to use fakeTimers. This arbitrary sleep will eventually become  flaky and also add a delay in our test runs

Reply:
Yeah I agree. but I've spent quite a while on this, nothing seemed to work not exactly sure why. 


This 250 is the duration of our moderate motion token, i can update it to reference that so that if we change the duration on bottomsheet it changes here also. 

Let's track this & I'll create a backlog issue for this. BottomSheet anyway needs better test with E2E

Reply:
Tracked here: https://github.com/razorpay/blade/issues/1168